src/plugins/vis_type_xy/public/utils/use_color_picker.tsx

```diff
@@ -61,8 +63,27 @@ export const useColorPicker = (
         onClose();
       };
 
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      const handleOutsideClick = useCallback(
+        (e: MouseEvent) => {
+          if (ref.current && !(ref.current! as any).contains(e.target)) {
+            onClose?.();
+          }
+        },
+        [onClose]
+      );
+
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      useEffect(() => {
+        document.addEventListener('click', handleOutsideClick);
+        return () => {
+          document.removeEventListener('click', handleOutsideClick);
+        };
+      }, [handleOutsideClick]);
```

Comment:
Is it possible for you to use EuiOutsideClickDetector instead of implementing your own? I ask because I noticed that there is some custom logic around finding ancestor elements in the EUI code:

https://github.com/elastic/eui/blob/master/src/components/outside_click_detector/outside_click_detector.ts

Reply:
Good point. Fixed here 587153d.

Reply:
@stratoula I think you could do the same in your pie chart color picker code.

Reply:
Nice! I will 😉